doc/TROUBLESHOOTING.md

```diff
@@ -145,3 +145,69 @@ Mountpoint logs should contain an error similar to below:
+If your storage provider does not support virtual style hosted bucket, you may recieve the following error:
+
+```
+Error: Failed to create S3 client
+
+Caused by:
+    0: initial ListObjectsV2 failed for bucket my-bucket in region us-east-1
+    1: Client error
+    2: Unknown CRT error
+    3: CRT error 1059: aws-c-io: AWS_IO_DNS_INVALID_NAME, Host name was invalid for dns resolution.
+Error: Failed to create mount process
+```
+
+In this case, try using `--force-path-style` CLI option when you are mounting the bucket using Mountpoint.
+
+NOTE - Third party storage provider are not officially supported by Mountpoint for Amazon S3.
+
+## Directory disappear after deleting all the files within it
+
+Amazon S3 does not support directories and objects are just grouped using prefix.
+Mountpoint automatically infers a directory structure for your bucket by treating the `/` separator after prefix in your object keys as a delimiter between directories.
+
+So, if all the files within a prefix is deleted, the prefix itself cease to exist.
+In this case, it is expected that mountpoint will not be able to show the directory for listing or other file system operation.
+
+Workaround to persist a directory could be creating a hidden empty file (for example, `.keep`).
+
+For more details on how Mountpoint maps S3 object keys to files and directories, see the [semantics documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#mapping-s3-object-keys-to-files-and-directories).
+
+## Input/output error after running workload for some time
+
+It is possible that your AWS credentials may get expired while your bucket is still mounted using Mountpoint.
+In that case you will get following errors for filesystem operations:
+
+```
+ls: reading directory '.': Input/output error
+```
+
+```
+cat: new_file.txt: Input/output error
+```
+
+Please check the Mountpoint logs. If you see the following errors, then your AWS credentials have expired.
+
+```
+[WARN] lookup{req=104 ino=1 name="Input"}:
+list_objects{id=20 bucket=plutodemo continued=false delimiter=/ max_keys=1 prefix=Input/}: mountpoint_s3_client::s3_crt_client:
+meta request failed duration=10.023623ms error=ClientError(Forbidden("The provided token has expired."))
+
+[WARN] lookup{req=106 ino=1 name="Input"}:
+head_object{id=21 bucket=plutodemo key=Input}: mountpoint_s3_client::s3_crt_client:
+meta request failed duration=11.087781ms error=ClientError(Forbidden("<no message>"))
+```
+
+You might need to have a fresh mount of your bucket with valid credentials.
+For more details to configure AWS credentials, see the [configuration documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#aws-credentials).
+
+## Throttling Errors
+
+When looking at the logs, these will appear as failed requests with `http_status=503` or `http_status=429` .
```

Comment:
Log examples, please! Also, as a developer apart from realising that S3 is throttling my app, what should I do? Will MP retry those automatically?

Reply:
The problem with log example here is that I have never experienced throttling from S3. I added it as it was there in support runbook. Probably, we should not be adding this case here.

Reply:
I see, so the main bit here is that Mountpoint does not do throttling itself. Thanks for describing a way of mitigating S3 throttling errors, but where does this come from? You mention there `3,500 PUT/COPY/POST/DELETE or 5,500 GET/HEAD`, these values may become outdated, I'd add a link to the source of that information.

Otherwise LGTM!